beacon_node/execution_layer/src/test_utils/execution_block_generator.rs

```diff
@@ -66,6 +70,37 @@ impl<T: EthSpec> Block<T> {
             },
         }
     }
+
+    pub fn as_execution_block_with_tx(&self) -> Option<ExecutionBlockWithTransactions<T>> {
+        match self {
+            Block::PoS(payload) => {
+                // We are not checking for transaction validity.
+                // So we use a default txs for testing purposes.
+                let txs = payload
+                    .transactions
+                    .iter()
+                    .map(|_| Transaction::default())
+                    .collect();
```

Comment:
I don't understand why we need this function, and why it's implementation is returning invalid transactions. If we were using it for payload reconstruction wouldn't we need this to return the correct transactions?

Reply:
The function is for payload reconstruction. I think we can just return an empty vector for the transactions field as we are doing in other functions in the `execution_block_generator` which returns an `ExecutionPayload`.
My comment does look confusing. I think I fixed it in [e9a50d1](https://github.com/sigp/lighthouse/pull/3292/commits/e9a50d126b845d2eeeae8092a018dba7cf147c84)

Reply:
yeah cool, that makes sense

thankfully we have some great payload reconstruction tests already ;)